### PR TITLE
Needs modifications

### PR DESCRIPTION
**Pull request by AnamariaaNistor**
- add and expose `status_updated_at `& `updated_by `
- update endpoints when status changes

### CODE REVIEW

File: app/api/help_seekers/needs_api.rb [hunk 1]
```diff
@@ -112,6 +114,7 @@ class NeedsApi < Grape::API
         post :close do
           need = current_user.my_needs.includes(:reviews).find(params[:id])
 
+          debugger
```
Comment (biancabordianu) [inline app/api/help_seekers/needs_api.rb]: I think you can delete this line.

File: app/api/help_seekers/needs_api.rb [hunk 2]
```diff
@@ -120,6 +123,7 @@ class NeedsApi < Grape::API
 
             need.reviews.create!(review_params)
             need.closed!
+            need.update!(status_updated_at: DateTime.now, updated_by: current_user.id)
```
Comment (biancabordianu) [inline app/api/help_seekers/needs_api.rb]: I think you can do only one update. WDYT?

File: app/api/volunteers/needs_api.rb [hunk 3]
```diff
@@ -83,6 +83,7 @@ class NeedsApi < Grape::API
               need.reviews.create!(review_params)
             end
 
+            need.update!(status_updated_at: DateTime.now, updated_by: current_user.id)
             need.completed!
```
Comment (biancabordianu) [inline app/api/volunteers/needs_api.rb]: I think you can do only one update. WDYT?

File: app/api/volunteers/needs_api.rb [hunk 4]
```diff
@@ -208,6 +211,7 @@ class NeedsApi < Grape::API
                 given_to_id: need.chosen_by.id
               )
 
+              need.update!(status_updated_at: DateTime.now, updated_by: current_user.id)
               need.reviews.create!(review_params)
               need.closed!
```
Comment (biancabordianu) [inline app/api/volunteers/needs_api.rb]: I think you can do only one update for need. WDYT?

File: app/api/help_seekers/entities.rb [hunk 5]
```diff
@@ -6,6 +6,8 @@ class Volunteer < Common::Entities::User
     class Need < Common::Entities::Need
       expose :chosen_by, using: Volunteer, expose_nil: true
       expose :reviews, using: Common::Entities::Review
+      expose :status_updated_at
+      expose :updated_by
```
Comment (petrucioata) [inline app/api/help_seekers/entities.rb]: We can expose them under the common namespace as they are useful for both volunteers and help seekers namespaces.

File: app/api/volunteers/entities.rb [hunk 6]
```diff
@@ -12,12 +12,16 @@ class HelpSeeker < Common::Entities::User
     class Need < Common::Entities::Need
       expose :added_by, using: HelpSeeker, expose_nil: true
       expose :reviews, using: Common::Entities::Review, expose_nil: true
+      expose :status_updated_at
+      expose :updated_by
```
Comment (petrucioata) [inline app/api/volunteers/entities.rb]: As mentioned for  the help seekers entities, is better to have them exposed inside common.

File: app/api/volunteers/entities.rb [hunk 7]
```diff
@@ -12,12 +12,16 @@ class HelpSeeker < Common::Entities::User
...
     end
 
     class RecommendedNeed < Need
       expose :contact_info
       expose :contact_phone_number
       expose :chosen_by, using: Volunteer, expose_nil: true
+      expose :status_updated_at
+      expose :updated_by
```
Comment (petrucioata) [inline app/api/volunteers/entities.rb]: Being the child class of `Need` it inherits all its fields, there's no need to expose them again, just expose those fields that are not part of the parent class, as is the case for contact information.

File: app/api/help_seekers/needs_api.rb (hunk 2 above)
Comment (petrucioata) [inline app/api/help_seekers/needs_api.rb]: Please take a look at the logs, my assumption is that these 2 lines will trigger 2 separate updates, we might consider having just one single update that includes the status update too.

File: app/api/volunteers/needs_api.rb (hunk 3 above)
Comment (petrucioata) [inline app/api/volunteers/needs_api.rb]: Same here, take a look please if there are 2 separate updates.

File: app/api/volunteers/needs_api.rb (hunk 4 above)
Comment (petrucioata) [inline app/api/volunteers/needs_api.rb]: Also here, let's try to use one single update for need and one create for review.

File: app/api/common/entities.rb [hunk 8]
```diff
@@ -16,6 +16,8 @@ class Need < Grape::Entity
       expose :id, documentation: { type: Integer }
       expose :description
       expose :status
+      expose :status_updated_at
+      expose :updated_by
```
Comment (petrucioata) [inline app/api/common/entities.rb]: Don't forget to set the type for them, please.

**Review by petrucioata — APPROVED** [review]
Well done! :clap:

**Review by biancabordianu — APPROVED** [review]

**Pull request merged**